Code/client/Games/Skyrim/Combat/CombatController.cpp

```diff
@@ -0,0 +1,91 @@
+#include "CombatController.h"
+#include "CombatTargetSelector.h"
+
+// TODO: ft
+
+void ArrayQuickSortRecursiveCombatTargets(GameArray<CombatTargetSelector*>* apArray, uint32_t aUnk3, uint32_t aUnk4)
+{
+    using TArrayQuickSort = void(GameArray<CombatTargetSelector*>* apArray, void* apFunction, uint32_t aUnk3, uint32_t aUnk4);
+    POINTER_SKYRIMSE(TArrayQuickSort, arrayQuickSort, 33285);
+
+    using TSortTargetSelectors = int64_t(int64_t, int64_t);
+    POINTER_SKYRIMSE(TSortTargetSelectors, sortTargetSelectors, 33282);
+
+    arrayQuickSort(apArray, sortTargetSelectors, aUnk3, aUnk4);
+}
+
+void CombatController::UpdateTarget()
+{
+    for (auto* pTargetSelector : targetSelectors)
+    {
+        if ((pTargetSelector->flags & 2) == 0)
+            pTargetSelector->Update();
+    }
+
+    if (targetSelectors.length > 1)
+        ArrayQuickSortRecursiveCombatTargets(&targetSelectors, 0, targetSelectors.length);
+
+    pActiveTargetSelector = nullptr;
+    BSPointerHandle<Actor> newTarget{};
+
+    if (targetSelectors.length)
+    {
+        for (auto* pTargetSelector : targetSelectors)
+        {
+            if ((pTargetSelector->flags & 1) != 0 && (pTargetSelector->flags & 2) == 0)
```

Comment:
if(pTargetSelector->flags & 3) == 1) ?

Reply:
The code in this function is a recreation of the original UpdateTarget function. The original code doesn't check for 3.